File: 01_projects/code/java-playwright/cbaps-dawms/basic-pom/src/test/java/com/dawms/pages/MilestoneStatusPage.java
package com.dawms.pages;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import com.microsoft.playwright.Locator;
import com.microsoft.playwright.Page;
import com.dawms.library.PlaywrightManager;

/**
 * MilestoneStatusPage: Final milestone and status validation page
 * Displays current submission milestone and status
 */
public class MilestoneStatusPage {

	private static final Logger log = LoggerFactory.getLogger(MilestoneStatusPage.class);
	private final Page page;
	private final PlaywrightManager pwm;

	private final Locator milestoneLabel;
	private final Locator statusLabel;

	public MilestoneStatusPage(Page page, PlaywrightManager pwm) {
		this.page = page;
		this.pwm = pwm;

		// Initialize locators
		this.milestoneLabel = page.locator("#milestone");
		this.statusLabel = page.locator("#status");

		// Stability anchor: wait for status to be visible
		pwm.waitVisible("#status");
		log.info("Milestone Status Page loaded successfully.");
	}

	/**
	 * Get current submission milestone
	 * @return Milestone text
	 */
	public String getMilestone() {
		String milestone = milestoneLabel.textContent();
		log.info("Retrieved milestone: " + milestone);
		return milestone;
	}

	/**
	 * Get current submission status
	 * @return Status text
	 */
	public String getStatus() {
		String status = statusLabel.textContent();
		log.info("Retrieved status: " + status);
		return status;
	}
}

File: 01_projects/code/python-playwright/cbaps-dawms/basic-pom/dawms/pages/milestone_status_page.py
"""
MilestoneStatusPage: Final milestone and status validation page
Displays current submission milestone and status
"""

import logging
from playwright.sync_api import Page, Locator
from dawms.library.playwright_manager import PlaywrightManager

logger = logging.getLogger(__name__)


class MilestoneStatusPage:
    """
    MilestoneStatusPage displays final milestone and status.
    This is the Python equivalent of the Java MilestoneStatusPage.
    """
    
    def __init__(self, page: Page, pwm: PlaywrightManager):
        """
        Initialize MilestoneStatusPage
        
        Args:
            page: Playwright Page object
            pwm: PlaywrightManager instance
        """
        self.page = page
        self.pwm = pwm
        
        # Locators
        self.milestone_label: Locator = page.locator("#milestone")
        self.status_label: Locator = page.locator("#status")
        
        # Stability anchor: wait for status to be visible
        pwm.wait_visible("#status")
        logger.info("Milestone Status Page loaded successfully.")
    
    def get_milestone(self) -> str:
        """
        Get current submission milestone
        
        Returns:
            Milestone text
        """
        milestone = self.milestone_label.text_content()
        logger.info(f"Retrieved milestone: {milestone}")
        return milestone
    
    def get_status(self) -> str:
        """
        Get current submission status
        
        Returns:
            Status text
        """
        status = self.status_label.text_content()
        logger.info(f"Retrieved status: {status}")
        return status
